aws/ec2.go

```diff
@@ -0,0 +1,84 @@
+package aws
+
+import (
+	"fmt"
+
+	awsgo "github.com/aws/aws-sdk-go/aws"
+	"github.com/aws/aws-sdk-go/aws/session"
+	"github.com/aws/aws-sdk-go/service/ec2"
+	"github.com/gruntwork-io/aws-nuke/logging"
+)
+
+var ec2Instances = make(map[string][]*string)
+
+// Build string that'll be shown in resources list
+func buildEntryName(instance ec2.Instance) string {
+	return fmt.Sprintf("ec2-%s-%s", *instance.InstanceId, *instance.InstanceType)
+}
+
+// Returns a formatted string of EC2 instance ids
+func getAllEc2Instances(session *session.Session, region string) ([]string, error) {
+	svc := ec2.New(session)
+
+	params := &ec2.DescribeInstancesInput{
+		Filters: []*ec2.Filter{
+			&ec2.Filter{
+				Name: awsgo.String("instance-state-name"),
+				Values: []*string{
+					awsgo.String("running"), awsgo.String("pending"),
+				},
+			},
+		},
+	}
+
+	output, err := svc.DescribeInstances(params)
+	if err != nil {
+		return nil, err
+	}
+
+	var entries []string
+	for _, reservation := range output.Reservations {
+		for _, instance := range reservation.Instances {
+			instanceID := *instance.InstanceId
+
+			attr, _ := svc.DescribeInstanceAttribute(&ec2.DescribeInstanceAttributeInput{
```

Comment:
Don't ignore errors. It's annoying to have `if err != nil` checks everywhere, but it's far more annoying when things fail silently!